ddsreplayer/test/blackbox/mcap/dds/HelloWorldDynTypesSubscriber.cpp

```diff
@@ -94,6 +87,7 @@ HelloWorldDynTypesSubscriber::HelloWorldDynTypesSubscriber(
 
 HelloWorldDynTypesSubscriber::~HelloWorldDynTypesSubscriber()
 {
+    data_->hz_msgs = data_->hz_msgs / (data_->n_received_msgs - 1);
```

Comment:
This is not thread safe (could be modified at the same time in a on_data_available), and a bit weird in my opinion (because you are reusing the attribute with multiple meanings).
I would have a cummulated time between messages attribute, and another with the mean, that is updated in fill_info with every message received.

Reply:
Additionally, this does not contemplate the case where no messages are received.